keanu-project/src/test/java/io/improbable/keanu/vertices/dbl/probabilistic/DirichletVertexTest.java

```diff
@@ -72,6 +119,36 @@ public void matchesMontrealDirichletLogPdf() {
             keanu.logPdf(new double[]{0.3, 0.4, 0.3}), 1e-6);
     }
 
+    @Test
+    public void logProbGraphMatchesMontrealDirichletLogPdf() {
+        DoubleVertex concentration = ConstantVertex.of(3., 4., 5.);
+        DirichletVertex vertex = new DirichletVertex(concentration);
+        DirichletDist baseline = new DirichletDist(new double[]{3, 4, 5});
+        LogProbGraph logProbGraph = vertex.logProbGraph();
+        LogProbGraphValueFeeder.feedValue(logProbGraph, concentration, concentration.getValue());
+
+        LogProbGraphValueFeeder.feedValue(logProbGraph, vertex, DoubleTensor.create(0.1, 0.6, 0.3));
+        double expectedDensity = Math.log(baseline.density(new double[]{0.1, 0.6, 0.3}));
+        LogProbGraphContract.matchesKnownLogDensity(logProbGraph, expectedDensity);
+
+        LogProbGraphValueFeeder.feedValueAndCascade(logProbGraph, vertex, DoubleTensor.create(0.3, 0.4, 0.3));
+        expectedDensity = Math.log(baseline.density(new double[]{0.3, 0.4, 0.3}));
+        LogProbGraphContract.matchesKnownLogDensity(logProbGraph, expectedDensity);
+    }
+
+
+    @Test(expected = GraphAssertionException.class)
```

Comment:
for exceptions that are thrown from your own code, please also assert on the message (using a `@Rule ExpectedException`)